compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -1698,6 +1708,40 @@ private void checkInLangLib(BLangInvocation iExpr, BType varRefType) {
         if (!langLibMethodExists) {
             dlog.error(iExpr.name.pos, DiagnosticCode.UNDEFINED_FUNCTION, iExpr.name.value);
             resultType = symTable.semanticError;
+            return;
+        }
+
+        checkIllegalStorageSizeChangeMethodCall(iExpr, varRefType);
+    }
+
+    private boolean isFixedLengthList(BType type) {
+        switch(type.tag) {
+            case TypeTags.ARRAY:
+                return (((BArrayType) type).state == BArrayState.CLOSED_SEALED);
```

Comment:
When we reach this point the state of inferred array types is always set as `CLOSED_SEALED` right? Or we'll have to check

```suggestion
                return (((BArrayType) type).state != BArrayState.UNSEALED);
```